dsc_lib/src/functions/min.rs

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use crate::DscError;
+use crate::configure::context::Context;
+use crate::functions::{AcceptedArgKind, Function};
+use serde_json::Value;
+use tracing::debug;
+
+#[derive(Debug, Default)]
+pub struct Min {}
+
+impl Function for Min {
+    fn min_args(&self) -> usize {
+        1
+    }
+
+    fn max_args(&self) -> usize {
+        usize::MAX
+    }
+
+    fn accepted_arg_types(&self) -> Vec<AcceptedArgKind> {
+        vec![AcceptedArgKind::Number, AcceptedArgKind::Array]
+    }
+
+    fn invoke(&self, args: &[Value], _context: &Context) -> Result<Value, DscError> {
+        debug!("min function");
+        if args.len() == 1 {
+            if let Some(array) = args[0].as_array() {
+                if array.len() > 1 {
+                    find_min(array)
+                }
+                else {
+                    Err(DscError::Parser("Array must contain more than 1 integer".to_string()))
+                }
+            }
+            else {
+                Err(DscError::Parser("List must contain more than 1 integer".to_string()))
+            }
+        }
+        else {
+            find_min(args)
+        }
+    }
+}
+
+fn find_min(args: &[Value]) -> Result<Value, DscError> {
+    let mut min_value = i64::MAX;
```

Comment:
Looks like you can just use built-in Rust function: https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.min